assets/js/src/admin/popup-editor/general.js

```diff
@@ -56,6 +56,79 @@
 					}
 				}
 			});
+
+			document.querySelector( '#pum-popup-settings-container' ).addEventListener( 'click', function(e) {
+				if ( Array.from(e.target.classList).includes('popup-type') || Array.from(e.target.parentElement.classList).includes('popup-type') ) {
+					const $container = jQuery( '#pum-popup-settings-container' );
+					if ( 1 === $container.length ) {
+						// Temporarily hardcoded object for testing...
+						const popupTypes = {
+							'center-popup': {
+								'size': 'medium',
+								'responsive_min_width': '0%',
+								'responsive_max_width': '100%',
+								'animation_type': 'fade',
+								'animation_speed': 350,
+								'location': 'center',
+								'overlay_disabled': false,
+								'stackable': false,
+								'disable_reposition': false,
+							},
+							'left-bottom-notice': {
+								'size': 'tiny',
+								'responsive_min_width': '0%',
+								'responsive_max_width': '100%',
+								'animation_type': 'slide',
+								'animation_speed': 500,
+								'animation_origin': 'left bottom',
+								'location': 'left bottom',
+								'position_bottom': 10,
+								'position_left': 10,
+								'position_fixed': true,
+								'overlay_disabled': true,
+								'stackable': true,
+								'disable_reposition': false,
+							},
+							'top-bar': {
+								'size': 'custom',
+								'custom_width': '100%',
+								'custom_height_auto': true,
+								'animation_type': 'fadeAndSlide',
+								'animation_speed': 300,
+								'animation_origin': 'center top',
```

Comment:
So when you want to animate in a straight line from an edge, always drop the second position.

- Slide straight up: animation origin = bottom.
- Slide straight down, origin = top.
- Slide straight left, origin = right

etc.